ivy/functional/ivy/nest.py

```diff
@@ -1199,18 +1217,28 @@ def nested_any(
             if ivy.any(fn(nest)):
                 return True
         for i, item in enumerate(nest):
-            if nested_any(item, fn, check_nests, False, extra_nest_types):
+            if nested_any(
+                item, fn, check_nests, check_attributes, False, extra_nest_types
+            ):
                 return True
         if check_nests and fn(nest):
             return True
     elif isinstance(nest, dict):
         for k, v in nest.items():
-            if nested_any(v, fn, check_nests, False, extra_nest_types):
+            if nested_any(
+                v, fn, check_nests, check_attributes, False, extra_nest_types
+            ):
                 return True
         if check_nests and fn(nest):
             return True
     elif fn(nest):
         return True
+    elif check_attributes and hasattr(nest, "__dict__"):
```

Comment:
maybe it'd be a little cleaner for this `elif` to come before the `elif fn(nest):` one.